### PR TITLE
[11.0][add] maintenance_equipment_hierarchy

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module allows to manage a hierarchy of equipments.

image

image

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
cc @nuriamartinxifre @ceeficent

**Review by astirpe — CHANGES_REQUESTED** [review]
In the equipment tree view or when searching equipments, the hierarchy is not displayed. Did you consider to define a def name_get() method?

_________



As improvement of the UX, I would move the field "Parent Equipment" at the top of the form (before "Equipment Category" or before "Name")

_________

Tested in Runbot:

- Open an already existing Equipment (eg.: A)
- click on button "Children"
- Create a new Equipment (eg.: B)

Result:
- B is created but is not a child of A

Expected:
- B is created as child of A

**Comment by astirpe** [comment]
Needs to be rebased to fix travis.

**Comment by LoisRForgeFlow** [comment]
Rebase done.

File: maintenance_equipment_hierarchy/views/maintenance_equipment_views.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <button name="preview_child_list"
+                        type="object"
+                        class="oe_stat_button"
+                        icon="fa-sitemap">
+                    <field string="Children" name="child_count" widget="statinfo"/>
+                </button>
+            </div>
+            <field name="location" position="before">
```
Comment (dalonsod) [inline maintenance_equipment_hierarchy/views/maintenance_equipment_views.xml:17]: As @astirpe recommended, the code could be

```
            <field name="category_id" position="before">
                <field name="parent_id" readonly="context.get('readonly_parent_id', False)"/>
            </field>

```

File: maintenance_equipment_hierarchy/models/maintenance_equipment.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        return True
+
+    def preview_child_list(self):
+        return {
+            'name': 'Child equipment of %s' % self.name,
+            'type': 'ir.actions.act_window',
+            'res_model': 'maintenance.equipment',
+            'res_id': self.id,
+            'view_mode': 'list,form',
+            'context': self.env.context,
```
Comment (dalonsod) [inline maintenance_equipment_hierarchy/models/maintenance_equipment.py:45]: As @astirpe recommended, the code could be

```
            'context': {
                **self.env.context,
                'default_parent_id': self.id,
                'readonly_parent_id': True},

```

Comment (dalonsod) [reply maintenance_equipment_hierarchy/models/maintenance_equipment.py:45]: @jbeficent @lreficent will you resume with this PR? If not, I'd continue with it

**Comment by dalonsod** [comment]
Superseded by #39

**Comment by astirpe** [comment]
@jbeficent is it okay for you to continue with the other PR and close this one?

**Pull request closed without merging**